.github/workflows/ci.yml

```diff
@@ -0,0 +1,368 @@
+        with:
+          glob: 'ethereum/referencetests/build/generated/sources/reference-test/**/*.java'
+          split-total: ${{ env.total-runners }}
+          split-index: ${{ matrix.runner-index }}
+      - name: run reference tests
+        run: ./gradlew ethereum:referenceTests:referenceTests $(echo ${{ steps.split-tests.outputs.test-suite }} | sed -e 's/^.*java\///' -e 's@/@.@g' -e 's/\.java//' -e 's/^/--tests /') --scan
+      - name: Upload Test Report
+        uses: actions/upload-artifact@v3
+        if: always() # always run even if the previous step fails
+        with:
+          name: junit-reference-${{ matrix.runner-index }}-test-results
+          path: '**/build/test-results/test/TEST-*.xml'
+          retention-days: 1
+  referenceTestQbft:
+    runs-on: ubuntu-latest
+    container: tmio/besu-build:1.0
+    needs: compile
+    timeout-minutes: 30
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 11
+          cache: gradle
+      - name: Cache classes
+        uses: actions/cache@v3
+        with:
+          path: "**/*.class"
+          key: ${{ runner.os }}-build-${{ github.sha }}
+      - name: qbft reference tests
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: consensus:qbft:referencetests --scan
+      - name: Upload Test Report
+        uses: actions/upload-artifact@v3
+        if: always() # always run even if the previous step fails
+        with:
+          name: junit-qbft-reference-test-results
+          path: '**/build/test-results/referenceTests/TEST-*.xml'
+          retention-days: 1
+  acceptanceTests:
+    runs-on: ubuntu-latest
+    needs: compile
+    timeout-minutes: 30
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 11
+          cache: gradle
+      - name: Cache classes
+        uses: actions/cache@v3
+        with:
+          path: "**/*.class"
+          key: ${{ runner.os }}-build-${{ github.sha }}
+      - name: pull test image
+        run: docker pull quorumengineering/acctests:latest
+      - name: ethereum acceptance tests
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: acceptanceTests --scan
+      - name: Upload Test Report
+        uses: actions/upload-artifact@v3
+        if: always() # always run even if the previous step fails
+        with:
+          name: junit-acceptance-test-results
+          path: '**/build/test-results/acceptanceTest/TEST-*.xml'
+          retention-days: 1
```

Comment:
What about publishing to artifactory?

Reply:
no idea how to go about doing that, prefer subsequent PR

Reply:
Publishing the jar libraries to https://hyperledger.jfrog.io/

I put this at the end of the file not to commet about acceptanceTests but to note that there isn't a step to push the jars to artifactory.